clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -560,13 +562,24 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
             log.trace("Received produce response from node {} with correlation id {}", response.destination(), correlationId);
             // if we have a response, parse it
             if (response.hasResponse()) {
+                // TODO: Sender should exercise PartitionProduceResponse rather than ProduceResponse.PartitionResponse
+                // https://issues.apache.org/jira/browse/KAFKA-10696
```

Comment:
nit: since we have the jira for tracking, can we remove the TODO? A few more of these in the PR.

Reply:
copy that